### PR TITLE
merge main to hocon / run eunit on GitHub actions

### PR DESCRIPTION
**Pull request by 6293**
review in conjunction with https://github.com/emqx/hocon/pull/53

### CODE REVIEW

File: test/cuttlefish_integration_test.erl [hunk 1]
```diff
@@ -60,6 +62,7 @@ all_the_marbles_test() ->
     Schema = cuttlefish_schema:file(tp("riak.schema")),
     Conf = [], %conf_parse:file(tp("riak.conf")),
     NewConfig = cuttlefish_generator:map(Schema, Conf),
+    io:format("asas ~p~n",[NewConfig]),
```
Comment (zmstone) [inline test/cuttlefish_integration_test.erl]: debug prints?

File: test/cuttlefish_integration_test.erl [hunk 2]
```diff
@@ -8,9 +8,11 @@
 generated_conf_file_test() ->
     {_, Mappings, _} = cuttlefish_schema:file(tp("riak.schema")),
     cuttlefish_conf:generate_file(Mappings, "../generated.conf"),
+    cuttlefish_conf:generate_file(Mappings, "./generatedaa.conf"),
     %% Schema generated a conf file, let's parse it!
-    Conf = cuttlefish_conf:file("../generated.conf"),
-    ?assertEqual("8099", proplists:get_value(["handoff","port"], Conf)),
+    io:format("sasasa ~p~n", [cuttlefish_conf:file("./generatedaa.conf")]),
```
Comment (zmstone) [inline test/cuttlefish_integration_test.erl]: debug prints ?

File: test/cuttlefish_escript_test.erl [hunk 3]
```diff
@@ -29,15 +29,15 @@
...
-        begin
-            ___OldLeader = group_leader(),
-            group_leader(cuttlefish_test_group_leader:new_group_leader(self()), self()),
-            try
-                __Forms
-            after
-                cuttlefish_test_group_leader:tidy_up(___OldLeader)
-            end
-        end).
+    (fun() ->
```
Comment (zmstone) [inline test/cuttlefish_escript_test.erl:32]: style nit, this file has indentation=4, the new code has 2

File: test/cuttlefish_integration_test.erl (hunk 1 above)
Comment (6293) [reply test/cuttlefish_integration_test.erl]: ty

**Comment by zmstone** [comment]
could you send a PR to `main` branch to add github action ?
then merge it to `hocon` branch.

**Review by zmstone — APPROVED** [review]

File: src/cuttlefish_datatypes.erl [hunk 4]
```diff
@@ -193,6 +193,16 @@ to_string(Value, MaybeExtendedDatatype) ->
...
 
+-spec to_quoted_string(term(), datatype()) -> string() | cuttlefish_error:error().
+to_quoted_string(V, Type) when Type =:= ip; Type =:= string;
+                               element(1, Type) =:= enum; element(1, Type) =:= duration;
+                               Type =:= bytesize; Type =:= file; Type =:= directory ->
+  quote(to_string(V, Type));
+to_quoted_string(V, Type) ->
+  to_string(V, Type).
+
+quote(Text) when is_list(Text) -> "\"" ++ Text ++ "\"".
```
Comment (zmstone) [inline src/cuttlefish_datatypes.erl:204]: maybe not an issue, but there is perhaps a need for quote escape here?
(Not sure if we use config generation at all)

**Comment by zmstone** [comment]
Good job.
It would have been better if the new commits were rebased instead

**Pull request merged**